spec/controllers/data_entry_access/users_controller_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe Admin::UsersController, type: :controller do
+  let!(:community) { create :community }
+  let!(:data_entry_user) { create :user, community: community, role: 'data_entry' }
+
+  before do
+    allow(controller).to receive(:authenticate_user!).and_return(true)
+    allow(controller).to receive(:current_user).and_return(data_entry_user)
+  end
+
+  describe "PUT #update" do
+    context 'when the user has an active time slot' do
+      before do
+        AccessTimeSlot.create(grantee_id: data_entry_user.id,
+          grantor_id: 100,
+          community_id: data_entry_user.community.id,
+          use: 'data_entry',
+          start_time: 1.hour.ago,
+          end_time: 1.hour.from_now
+        )
+      end
+      let!(:user) {create :user, community: community, role: "admin", first_name: "Elizabeth"}
+      it 'does NOT update user role but does update other attributes' do
+        user.first_name = "Jane"
+        user.role = "data_entry"
+        put :update, params: {
+          community_slug: community.slug, 
+          id: user.id, 
+          user: user.attributes
+        }
+        expect(user.reload.role).to eq "admin"
+        expect(user.reload.first_name).to eq "Jane"
```

Comment:
I'm about to open an issue for this, but the `reload` bit here differs from the pattern in other controller tests testing #update actions. It's my belief that this is necessary, because otherwise we're just writing a test that is basically

item.attribute = "banana"
expect(item.attribute).to eq "banana"

The reload makes sure it's not just the in memory object that's changed.